File: frontend/src/api/menu.ts
import request from '../utils/request';
import type { MenuItem } from '../types/menu';

/**
 * Menu API definitions
 */

// Get menu tree for current user
export const getMenuTree = (): Promise<MenuItem[]> => {
  return request.get('/menu/tree');
};

// Get all menus (admin only)
export const getAllMenus = (): Promise<MenuItem[]> => {
  return request.get('/menu/all');
};

// Get menu by ID
export const getMenuById = (id: number): Promise<MenuItem> => {
  return request.get(`/menu/${id}`);
};

// Create menu
export interface CreateMenuRequest {
  parentId: number;
  path: string;
  name: string;
  component: string;
  sort: number;
  meta: {
    icon: string;
    title: string;
    hidden: boolean;
    keepAlive: boolean;
  };
  btnPerms?: string[];
}

export const createMenu = (data: CreateMenuRequest): Promise<MenuItem> => {
  return request.post('/menu', data);
};

// Update menu
export interface UpdateMenuRequest {
  id: number;
  parentId?: number;
  path?: string;
  name?: string;
  component?: string;
  sort?: number;
  meta?: {
    icon?: string;
    title?: string;
    hidden?: boolean;
    keepAlive?: boolean;
  };
  btnPerms?: string[];
}

export const updateMenu = (data: UpdateMenuRequest): Promise<MenuItem> => {
  return request.put(`/menu/${data.id}`, data);
};

// Delete menu
export const deleteMenu = (id: number): Promise<void> => {
  return request.delete(`/menu/${id}`);
};

File: backend/router/system/menu.go
package system

import (
	"k-admin-system/api/v1/system"
	"k-admin-system/middleware"

	"github.com/gin-gonic/gin"
)

// InitMenuRouter 初始化菜单路由
func InitMenuRouter(router *gin.RouterGroup) {
	menuApi := system.MenuApi{}

	// 受保护的路由（需要JWT认证和Casbin授权）
	protectedGroup := router.Group("/menu")
	protectedGroup.Use(middleware.JWTAuth())
	protectedGroup.Use(middleware.CasbinAuth())
	{
		// 菜单CRUD操作
		protectedGroup.POST("", menuApi.CreateMenu)
		protectedGroup.PUT("", menuApi.UpdateMenu)
		protectedGroup.DELETE("/:id", menuApi.DeleteMenu)
		protectedGroup.GET("/:id", menuApi.GetMenu)
		protectedGroup.GET("/all", menuApi.GetAllMenus)
	}

	// 菜单树查询（仅需要JWT认证，不需要Casbin授权）
	// 因为该接口根据roleId过滤菜单，已经实现了权限控制
	menuTreeGroup := router.Group("/menu")
	menuTreeGroup.Use(middleware.JWTAuth())
	{
		menuTreeGroup.GET("/tree", menuApi.GetMenuTree)
	}
}
